### PR TITLE
Guarin lig 596 add dinoloss

### PR DESCRIPTION
**Pull request by guarin**

### CODE REVIEW

File: lightly/loss/dino_loss.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        super().__init__()
+        self.warmup_teacher_temp_epochs = warmup_teacher_temp_epochs
+        self.teacher_temp = teacher_temp
+        self.student_temp = student_temp
+        self.center_momentum = center_momentum
+        
+        self.register_buffer("center", torch.zeros(1, out_dim))
+        # we apply a warm up for the teacher temperature because
+        # a too high temperature makes the training instable at the beginning
+        self.teacher_temp_schedule = np.linspace(
```
Comment (philippmwirth) [inline lightly/loss/dino_loss.py:104]: could use torch linspace here over numpy

File: tests/loss/test_DINOLoss.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+import numpy as np
+import torch
+from torch import nn
+import torch.nn.functional as F
+
+from lightly.loss import dino_loss
+from lightly.loss import DINOLoss
+
+
+class FacebookDINOLoss(nn.Module):
```
Comment (philippmwirth) [inline tests/loss/test_DINOLoss.py:13]: maybe add a comment that you copy pasted this from somewhere so it's very clear what the purpose is

**Review by philippmwirth — COMMENTED** [review]
Looks mostly good 👍

File: lightly/loss/dino_loss.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    """Concatenates multiple outputs from the student model into a single 
+    tensor.
+    
+    Args:
+        batch_size:
+            The batch size B where B = number of images.
+        out_dim:
+            The dimensions of the model output.
+        student_out:
+            A single tensor with shape (B * V, D) or a list of tensors where
```
[the remark this answers is not in the document]
Comment (guarin) [reply lightly/loss/dino_loss.py:23]: Yes it is the number of views. Added a comment about it.

File: lightly/loss/dino_loss.py (hunk 1 above)
Comment (guarin) [reply lightly/loss/dino_loss.py:104]: Ah good point!

**Review by philippmwirth — APPROVED** [review]

File: tests/loss/test_DINOLoss.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
```
Comment (IgorSusmelj) [inline tests/loss/test_DINOLoss.py:13]: Would drop the `Facebook` part here. Or are there various implementations of the same loss?

**Review by philippmwirth — APPROVED** [review]
Looks good!
